db/src/db.rs

```diff
@@ -390,6 +393,24 @@ pub fn read_db(conn: &rusqlite::Connection) -> Result<DB> {
     Ok(DB::new(partition_map, schema))
 }
 
+use itertools;
```

Comment:
Is it common to have these `use` and `type` statements in the middle of a file? Also, in general, this is getting pretty long, maybe break it out into a new file?